base/strings/io.jl

```diff
@@ -382,12 +382,14 @@ julia> unescape_string("\\\\101") # octal
 ## See also
 [`escape_string`](@ref).
 """
-function unescape_string(io, s::AbstractString)
+function unescape_string(io, s::AbstractString, keep_esc::AbstractArray{<:AbstractChar}=Char[])
```

Comment:
This adds an argument to an exported function, so it should be documented. Also I'd simply call this `keep`.

Finally, it could be worth using `nothing` or `()` as a default value to avoid introducing an overhead when that argument is omitted. Not sure whether it makes a significant difference, though.

Reply:
Did you mean `unescape_string(io, s::AbstractString, keep = nothing)` ? I don't think that should be used otherwise it will throw `MethodError` in place of `ArgumentError`.